Configuration/Geometry/python/GeometryExtended2023D45_cff.py

```diff
@@ -5,7 +5,7 @@
 
 from Geometry.CMSCommonData.cmsExtendedGeometry2023D45XML_cfi import *
 from Geometry.TrackerNumberingBuilder.trackerNumberingGeometry_cfi import *
-from SLHCUpgradeSimulations.Geometry.fakeConditions_phase2TkT14_cff import *
+from SLHCUpgradeSimulations.Geometry.fakeConditions_phase2TkT15_cff import *
```

Comment:
Please note that directly changing this file is forbidden. The changes should be propagated into [dict2023Geometry.py](https://github.com/cms-sw/cmssw/blob/master/Configuration/Geometry/python/dict2023Geometry.py) and the corresponding script [generate2023Geometry.py](https://github.com/cms-sw/cmssw/blob/master/Configuration/Geometry/scripts/generate2023Geometry.py) should be run to regenerate the geometry files.

Reply:
Hi @kpedro88 I am not familiar with the procedure for updating these files. 
For my own education why is it forbidden to change it? 
Also it is not clear to me what does it mean to regenerate the geometry files? Do you have instructions on how to run this script? 
By the way I am not changing the geometry, just providing a different set of conditions.

Reply:
@mmusich because you have edited this file by hand, if someone were to add a new phase 2 detector version using the T15 tracker, they would not pick up this change and would get incorrect results. In fact, there is a unit test that enforces this, so this PR would not pass the test. The file also states at the top:
https://github.com/cms-sw/cmssw/blob/4590f5d4ba0a20f4fc13295e2d376820cab49d9e/Configuration/Geometry/python/GeometryExtended2023D45_cff.py#L3-L4

The generate2023Geometry.py script generates the following files for each phase 2 detector version:
https://github.com/cms-sw/cmssw/blob/f46916dae7609d7808eb6b9d4c9a996c5dcf981b/Configuration/Geometry/scripts/generate2023Geometry.py#L39-L41

It does this using the information in dict2023Geometry.py, to maintain the consistency. (If you checkout the packages Configuration/Geometry and Geometry/CMSCommonData in advance, it will even put the files in the right places, so you just have to `git add` them and then commit.)

Reply:
@kpedro88,  so just for my understanding, the correct way to change anything in a phase-2 workflow is to create a new an entirely new `D*` version by using the scripts you point above?

Reply:
If the actual detector geometry is changed, indeed a new subdetector version should be added, and then a new D* full detector version that uses the new subdetector.

In this case, you're not changing the geometry, but adding some related information. New D* versions are not necessary in that case, but updating the component files is still not allowed. Only [dict2026Geometry.py](https://github.com/cms-sw/cmssw/blob/master/Configuration/Geometry/python/dict2026Geometry.py) can be updated, and then you have to rerun [generate2026Geometry.py](https://github.com/cms-sw/cmssw/blob/master/Configuration/Geometry/scripts/generate2026Geometry.py) with the existing D* versions to update the generated files. (While this is not the most automatic procedure, it was the only agreement that could be made between all relevant focus areas to keep things consistent, and it is enforced by a unit test.)